### DISCUSSION TITLE
Setup for Windows/Linux Clients for non-standard port for connections

### DISCUSSION THREAD

**Discussion by stimpy-ch** [strongswan/strongswan#546] [Q&A] [answered]
Hi all

I did setup the strongswan according to [a tutorial](https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-20-04). Additionally to that I did also add to my /etc/ipsec.conf the following line:
`leftikeport=<otherPort>`

While the configuration on Android client was straight forward, I do not manage to get the clients working on Windows and Kubuntu. I fear that the only issue is that I don't manage to set the non standard port.

Can anyone tell me if this will be possible and how?

Many thanks

**Comment by tobiasbrunner** [comment, marked as answer]
> Additionally to that I did also add to my /etc/ipsec.conf the following line:
> `leftikeport=<otherPort>`

That does not actually change the listening port (it's only used when initiating a connection and arbitrary ports only work with the _socket-dynamic_ plugin). To change the listening port you have to change _charon.port_nat_t_ (see [this wiki page](https://wiki.strongswan.org/projects/strongswan/wiki/NatTraversal#Custom-Server-Ports) for details).

> I fear that the only issue is that I don't manage to set the non standard port.

If a client does not allow you to configure a custom remote port, this can't work. Windows definitely does not support it. And while our NM plugin (used in GNOME) supports this since 1.5.0, KDE uses their own plugin for the Plasma desktop, which I guess does not. However, you might be able to manually set the port in the NM connection file (`server-port` option, which requires charon-nm from strongSwan 5.8.3 or newer to work).

**Reply by stimpy-ch** [reply]
Thank you very much Tobias Brunner. I see I cannot do what I intended to do with the clients I have.
